Adds MIDI source support

HunterTheCraft-OFICIAL opened the pull request:
## Pull Request: Add MIDI Font Support

This pull request introduces a new font called "midi" to the cfonts library. This font is designed to fit within three lines and is based on the existing "tiny" font, with adjustments made for size and spacing.

### Modifications

The following files were modified:

- `font_test.rs`
- `args_test.rs`
- `font.rs`
- `config.rs`
- `args.rs`
- `constants.js`

### New Files

The following file was added:

- `midi.json`

### Additional Notes

- The `end-to-end_test.rs` file was not modified as I require further guidance on the necessary changes.
- The "midi" font name is open to suggestions. If you have a preferred name, please let me know.

### Branch Name Suggestion

- `feat/add-midi-font`

Thank you for your time and consideration. I look forward to your feedback and am happy to make any further adjustments.

HunterTheCraft-OFICIAL committed:
Adds MIDI source support

dominikwilkowski commented on fonts/midi.json:
```diff
@@ -0,0 +1,306 @@
+	"name": "midi",
+	"version": "0.2.0",
+	"homepage": "https://github.com/dominikwilkowski/cfonts",
+	"colors": 2,
```
There seems to be only 1 color present in your chars

dominikwilkowski commented on fonts/midi.json:
```diff
@@ -0,0 +1,306 @@
+            "█▀█"
+        ],
+		"B": [
+			"█▄▄",
+            "█ █",
```
Some mixed indentation worth fixing

dominikwilkowski commented on fonts/midi.json:
```diff
@@ -0,0 +1,306 @@
+{
+	"name": "midi",
+	"version": "0.2.0",
```
I would start with `0.1.0` since this is an addition

dominikwilkowski commented on rust/src/config.rs:
```diff
@@ -27,6 +27,8 @@ pub enum Fonts {
 	FontChrome,
 	/// ![The "huge" cfonts font](https://raw.githubusercontent.com/dominikwilkowski/cfonts/released/img/huge.png)
 	FontHuge,
+	/// ![The "midi" cfonts font](https://raw.githubusercontent.com/dominikwilkowski/cfonts/released/img/midi.png)
```
You're adding a link to a screenshot here but not the actual screenshot

dominikwilkowski reviewed (changes requested):
Thanks for the submission.
I left some comments.
Apart from making CI pass and addressing the comments may I ask what the difference is between `tiny` and `midi`? Perhaps that missing screenshot will help. (note the size of all screenshots)

HunterTheCraft-OFICIAL committed:
Format Rust code, update midi.json and yarn.lock

HunterTheCraft-OFICIAL committed:
Fixes Rust paths and formatting, adds Midi image

HunterTheCraft-OFICIAL commented:
> Obrigado pelo envio. Deixei alguns comentários. Além de fazer o CI passar e abordar os comentários, posso perguntar qual é a diferença entre `tiny`e `midi`? Talvez essa captura de tela ausente ajude. (observe o tamanho de todas as capturas de tela)



> Thanks for the submission. I left some comments. Apart from making CI pass and addressing the comments may I ask what the difference is between `tiny` and `midi`? Perhaps that missing screenshot will help. (note the size of all screenshots)

ok I present the differences between tiny and midi fonts

To the letters in tiny and midi fonts 
![Image](https://github.com/user-attachments/assets/5e5de78c-b49f-4d58-8798-518eca884f40)


the numbers in tiny and midi fonts
![Image](https://github.com/user-attachments/assets/fff2f188-5dc1-46f5-ae7d-24f816828814)


I have already made some of the requests but not all of them have been received yet.

dominikwilkowski commented:
Hey @HunterTheCraft-OFICIAL 

Looking at the difference between both `tiny` and your `midi` I'm not sure it's worth adding to the library.
If you got a font that's more different feel free to add it but this just seems to be the same font just a row longer.
Perhaps you could use different blocks to get to a different look while retaining the stretch? There are a couple [block elements](https://en.wikipedia.org/wiki/Block_Elements) to choose from.

dominikwilkowski commented:
Closing this for now. Feel free to open another PR when you find a new font.
